src/MessageReader.test.ts

```diff
@@ -520,4 +520,16 @@ module builtin_interfaces {
 
     expect(read).toEqual({ status: 2 });
   });
+
+  it.each(["wstring field", "wstring[] field"])(
+    "should throw exepction when encountering wstring fields",
+    (msgDef) => {
+      const buffer = Uint8Array.from(Buffer.from("00010000000000007b000000", "hex"));
+      const reader = new MessageReader(parseMessageDefinition(msgDef, { ros2: true }));
+
+      expect(() => reader.readMessage(buffer)).toThrow(
+        "wstring is implementation-defined and therefore not supported",
```

Comment:
Where is this information coming from? 
Looking at [this](https://design.ros2.org/articles/wide_strings.html), it looks like they should just use UTF-16?

Reply:
https://design.ros2.org/articles/wide_strings.html#size-of-a-wide-string

> Some DDS implementations currently use 32bit types to store wide strings values. This may be due to DDS-XTYPES 1.1 section 7.3.1.5 specifying wchar as a 32bit value. However this changes in DDS-XTYPES 1.2 section 7.3.1.4 to be a 16bit value. It is expected that most DDS implementations will switch to 16bit character storage in the future. ROS 2 will aim to be compatible with DDS-XTYPES 1.2 and use 16bit storage for wide characters. Generated code for ROS 2 messages will automatically handle the conversion when a message is serialized or deserialized.